### PR TITLE
:sparkles: Add a --version option to controller-gen

### PR DESCRIPTION
**Pull request by dlespiau**
It's useful to know the version of a `controller-gen` binary lying around. We could write checks in the kubebuilder `Makefile` to upgrade `controller-gen` when needed, or at least warn about it.

Fixes: #259 

I checked that this indeed worked with main modules when using `go get`: https://github.com/dlespiau/go-main-module-version.

### CODE REVIEW

File: pkg/version/version.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		return "", errors.New("binary has not been built with module support")
+	}
+	return info.Main.Version, nil
+}
+
+// Print prints the main module version on stdout
+func Print() error {
+	v, err := version()
+	if err != nil {
+		return err
```
Comment (droot) [inline pkg/version/version.go]: Looks like nothing will get printed for the end-user. We should probably print `Version: Unknown` in this case.

**Comment by droot** [comment]
Also, can you pl. update the commit and PR as per our template --> https://raw.githubusercontent.com/kubernetes-sigs/controller-tools/master/.github/pull_request_template.md

**Comment by droot** [comment]
I took another look and feel that we can simplify the version function by not returning error. In case of error, we simply return "unknown" version.

**Comment by dlespiau** [comment]
The error was propagated and displayed by the main program explaining why the version coud not be displayed (not compiled with go modules). But I see the point: may as well always display a "Version: .*" string that can be parsed in always the same way.

One detail is that the go runtime with return version `(devel)` when compiled from a local checkout. I chose to follow that lead and display `(unknown)` instead of the suggested `Unknown` so we have a stronger invariant. Also documented all the possible behaviours of `Print`.

**Comment by droot** [comment]
>One detail is that the go runtime with return version (devel) when compiled from a local checkout. I chose to follow that lead and display (unknown) instead of the suggested Unknown so we have a stronger invariant. Also documented all the possible behaviours of Print.

Sounds good to me.

**Pull request merged**